python/cudf/cudf/core/column/column.py

```diff
@@ -2329,3 +2344,10 @@ def concat_columns(objs: "MutableSequence[ColumnBase]") -> ColumnBase:
 
     # Filter out inputs that have 0 length, then concatenate.
     return libcudf.concat.concat_columns([o for o in objs if len(o)])
+
+
+def _has_any_nan(arbitrary):
+    return any(
+        (isinstance(x, (np.floating, float)) and np.isnan(x))
```

Comment:
I ran [a microbenchmark](https://github.com/bdice/python_microbenchmarks/blob/main/microbenchmarks/numpy_floating.py) and the suggestion below appears to be fastest for `float` or `np.float64` data. For `float32` data, the impact is very slightly negative, and for `str` / non-float data, there is no impact. Feel free to accept or reject this suggestion based on your assessment of what data types are likely to hit this path.
```suggestion
        ((isinstance(x, float) or isinstance(x, np.floating)) and np.isnan(x))
```

Raw benchmarks for `float` data (the `float` and `np.floating` standalone checks are for comparison):
```
isinstance(x, float): 0.00256905
isinstance(x, float) or isinstance(x, np.floating): 0.00304044
isinstance(x, np.floating): 0.00706707
isinstance(x, (float, np.floating)): 0.00778381
isinstance(x, (np.floating, float)): 0.0115429
```